src/smart.rs

```diff
@@ -181,13 +201,16 @@ impl Heap {
 
 /// `Block` type returned by `SmartAllocator`.
 #[derive(Debug)]
-pub struct SmartBlock<M>(CombinedBlock<M>, usize);
+pub struct SmartBlock<M: Debug + Any, B: Block<Memory = M> = CombinedBlock<M>>(
```

Comment:
Why `Any`?

Reply:
Why `M` at all. You are able to calculate it from `B` alone.

Reply:
The idea is to keep the old usage of `SmartAllocator<B>`, so it can still be used as such. Is this unnecessary?

Reply:
I understand that. No. I think this is unnecessary for `SmartBlock`